api/proto/teleport/legacy/types/types.proto

```diff
@@ -6451,8 +6451,15 @@ message PluginEntraIDSyncSettings {
   // DefaultOwners are the default owners for all imported access lists.
   repeated string default_owners = 1;
 
-  // SSOConnectorID is the name of the Teleport SSO connector created and used by the Entra ID plugin
+  // SSOConnectorID is the name of the Teleport SSO connector created and used by the Entra ID plugin.
   string sso_connector_id = 2;
+
+  // use_system_credentials uses the system available credentials instead of using the OIDC flow to authenticate
+  // to Azure EntraId. This is important for clusters that are not available through internet.
+  bool use_system_credentials = 3;
```

Comment:
I don't have much context here, but should we consider the usual anti-enum advice?

* https://protobuf.dev/programming-guides/api/#never-use-booleans-for-two-states
* https://cloud.google.com/apis/design/design_patterns#bool_vs_enum_vs_string

Reply:
Changed